console_backend/src/ntrip_tab.rs

```diff
@@ -339,45 +291,33 @@ impl NtripState {
     pub fn connect(
         &mut self,
         msg_sender: MsgSender,
-        url: String,
-        username: String, // empty username is None
-        password: String,
-        gga_period: u64,
-        pos: Option<(f64, f64, f64)>,
+        mut heartbeat: Heartbeat,
+        options: NtripOptions,
     ) {
-        if self.connected_thd.is_some() {
+        if self.connected_thd.is_some() && heartbeat.get_ntrip_connected() {
             // is already connected
             return;
         }
 
-        let pos_mode = pos
-            .map(|(lat, lon, alt)| PositionMode::Static { lat, lon, alt })
-            .unwrap_or(PositionMode::Dynamic);
-
-        let options = NtripOptions {
-            url,
-            pos_mode,
-            username: Some(username).filter(|s| !s.is_empty()),
-            password: Some(password).filter(|s| !s.is_empty()),
-            nmea_period: gga_period,
-            client_id: "00000000-0000-0000-0000-000000000000".to_string(),
-        };
-
         self.options = options.clone();
         let last_data = self.last_data.clone();
         self.set_running(true);
         let running = self.is_running.clone();
+        heartbeat.set_ntrip_connected(true);
+        println!("connected");
```

Comment:
Use logger

Reply:
did we want to include ntrip connect to logger or just when it errors?